components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/model/BuildTimeMetrics.java

```diff
@@ -0,0 +1,87 @@
+package com.gradle.enterprise.model;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+
+public class BuildTimeMetrics {
+
+    private final Duration initialBuildTime;
+    private final Duration instantSavings;
+    private final Duration instantSavingsBuildTime;
+    private final Duration pendingSavings;
+    private final Duration pendingSavingsBuildTime;
+
+    private BuildTimeMetrics(
+            Duration initialBuildTime,
+            Duration instantSavings,
+            Duration instantSavingsBuildTime,
+            Duration pendingSavings,
+            Duration pendingSavingsBuildTime) {
+        this.initialBuildTime = initialBuildTime;
+        this.instantSavings = instantSavings;
+        this.instantSavingsBuildTime = instantSavingsBuildTime;
+        this.pendingSavings = pendingSavings;
+        this.pendingSavingsBuildTime = pendingSavingsBuildTime;
+    }
+
+    public static BuildTimeMetrics from(BuildValidationData firstBuild, BuildValidationData secondBuild) {
+        final Duration buildTimeFirstBuild = firstBuild.getBuildTime();
+        final Duration buildTimeSecondBuild = secondBuild.getBuildTime();
+        final TaskExecutionSummary executedCacheableTaskSummarySecondBuild = secondBuild.getExecutedCacheableSummary();
+        final BigDecimal serializationFactorSecondBuild = secondBuild.getSerializationFactor();
+
+        if (buildTimeFirstBuild == null || buildTimeSecondBuild == null || executedCacheableTaskSummarySecondBuild == null || serializationFactorSecondBuild == null) {
+            return null;
+        }
```

Comment:
I'm not sure about this `null` checking. It's unclear if the Gradle Enterprise API could return these as `null`. We also we deliberately return [these fields as `null` from the API client](https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob/3870d12078a7410dceb439b53085ebed6cb63c98/components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/cli/FetchBuildValidationDataCommand.java#L101-L114) if anything goes wrong.

Reply:
`It's unclear if the Gradle Enterprise API could return these as null.` Did you check the OpenAPI definition?

Reply:
https://docs.gradle.com/enterprise/api-manual/#reference_documentation

Reply:
I did check it. We are already doing some `null` checking anyway on the response fields. It would be nice to isolate the `null` checking to one place or remove it completely.

Reply:
Can't we just remove these null checks here? These values should never be null by contract.

Reply:
> Can't we just remove these null checks here? These values should never be null by contract.

I think we came to a conclusion to this [in the other thread](https://github.com/gradle/gradle-enterprise-build-validation-scripts/pull/383#discussion_r1152662319).